common/gossmap.c

```diff
@@ -176,25 +176,25 @@ u32 gossmap_chan_idx(const struct gossmap *map, const struct gossmap_chan *chan)
 	return chan - map->chan_arr;
 }
 
-/* htable can't handle NULL values, so we add 1 */
+/* htable can't handle NULL or 1 values, so we add 2 */
```

Comment:
I maybe have missed your explanation in this bug fixing for ccan hash map, I think it is related to the memory pointer? that the hash map use like key?

Just curious on why the table can handle these values :)

Reply:
It uses these values to mean "empty" and "deleted".  Now, usually it's OK, because it also puts hash bits in unused bits (bits which are the same for every value in the map), and there are usually lots of these: values are usually all heap pointers.  But it's *possible* that the hash bits are all 0, in which case these would look like empty or deleted buckets.

It's always been possible, but the updated htable added an assertion to check for (void *)1.